store/models/orm.go

```diff
@@ -228,6 +229,32 @@ func (orm *ORM) BridgeTypeFor(name string) (BridgeType, error) {
 	return tt, err
 }
 
+// MarkRan will set Ran to true for a given initiator
+func (orm *ORM) MarkRan(i *Initiator) error {
+	dbtx, err := orm.Begin(true)
+	if err != nil {
+		return err
+	}
+	defer dbtx.Rollback()
+
+	var ir Initiator
+	if err := orm.One("ID", i.ID, &ir); err != nil {
```

Comment:
If you already have the initiator with `i *Initiator`, why do you retrieve it again?